components/vf-banner/vf-banner--states.scss

```diff
@@ -0,0 +1,43 @@
+.vf-banner--alert {
+  .vf-banner__content {
+    display: flex;
+  }
+  .vf-banner__text,
+  [class*='vf-text'] {
+    color: currentColor;
+  }
+}
+
+.vf-banner--success {
+  background-color: set-color(vf-color--green--light);
+  color: set-ui-color(vf-ui-color--black);
+}
+
+.vf-banner--alert {
```

Comment:
As .vf-banner--alert is the "general" class, it should come first in the Sass or else there are specificity issues:

![image](https://user-images.githubusercontent.com/928100/78368696-4cc15580-75c4-11ea-83bf-7d2d74687efb.png)

Reply:
kinda related(?): I was wondering if we should phase out (don't laugh) `vf-banner--phase` and instead have it as `vf-banner--alert vf-banner--info` or something.

Reply:
because we need a generic class for the padding (which `vf-banner--phase` has) but I think it's confusing to have `vf-banner--phase vf-banner--success`

Reply:
I've included a `vf-banner--info` variant. Changed `vf-banner--alert vf-banner--alert` duplication so the red one is `vf-banner--danger`

Reply:
That makes sense. 

On the `:hover` `:focus` I was thinking more accessibility if you miss the animation while tabbing or hovering, something like:

(tab first, then hover)

![Screen Shot 2020-04-03 at 21 48 48](https://user-images.githubusercontent.com/928100/78399568-68911f80-75f5-11ea-98fc-4135ba0a5608.gif)

```css
.vf-banner__button:focus {
    border: 1px solid white;
}

.vf-banner__button:hover {
    border-bottom: 1px solid white;
}
```

Reply:
ok, so maybe for `:focus` but that border on `:hover` is a little EBI/VF1.x—y

Also - I’ll look at adding the text-crop to `vf-core` so we can make the box actually square for `:focus` because it looks a little ugly.

If we _really_ need something like a border bottom on `:hover` perhaps we need it to look more like a button than it does now (it has no styling from researching several design systms alerts styles).

I’ll try and do both - the text-crop so the square outline is actually _square_ and also an example of it being a button, perhaps something that’s inset - but that kinda goes against current aesthetics.

Reply:
I mean, in saying some of that - the hover effect isn’t really part of the system either :sob: 

I’m sure there’s a good compromise out there :)

Reply:
Thinking about this, why do we _have_ to use an icon? Just as well or better to use "Close" or "Hide". 

For example, opening www.gov.uk in an incognito. It's a longer notification, but (as usual) their use of text works better most other systems.

The text is only a few characters longer (plus more explicit) and will bypass issues discussed above -- it will also reinforce a principle of avoiding icons wherever possible.

Reply:
I'm sure a banner will take be larger on a small viewport, and there would be less text for 'hide' rather than a `x` icon. 

re: hover/focus.

As it's a button there's the cursor change on hover (although touch devices will have to guess).

I've tidied up the `:focus` state so that it's a nice square box.

Reply:
Probably more than fair for now. As we don't even have a real use case for dismissing notifications (beyond the we-probably-need-it from HCA), we're at risk of premature optimisation.